### PR TITLE
Setup GH actions

### LINKED ISSUE #20: Automated testing running in GH Actions
Follow-up to #19

### PR DESCRIPTION
**Pull request by RageZBla**
Resolves #20 

# What changed

- Add new GH action to run cypress e2e test

# Testing instructions

Just check github actions results

### CODE REVIEW

**Comment by RageZBla** [comment]
@ann-kilzer can you take a first look? If it looks OK to you, I would clean it up.

File: .github/workflows/eslint.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # More details at https://github.com/eslint/eslint
 # and https://eslint.org
 
-name: ESLint
```
Comment (ermish) [inline .github/workflows/eslint.yml:6]: Can you undo this change? There will be multiple linters, so it is better to be explicit

File: package.json [hunk 2]
```diff
@@ -34,6 +35,7 @@
         "eslint-plugin-vue": "^9.2.0",
         "husky": "^8.0.1",
         "jest": "^29.1.2",
+        "jest-environment-jsdom": "^29.2.1",
```
Comment (ermish) [inline package.json:38]: Can we move these package.json changes into a separate pull request? 
Generally, it's better to tie to the actually dependencies changed that rely on it. Also, we should review what they're used for.

**Review by ermish — CHANGES_REQUESTED** [review]
Left a few changes. Please take a look when you get a chance. Thanks for contributing!

File: .github/workflows/eslint.yml (hunk 1 above)
Comment (RageZBla) [reply .github/workflows/eslint.yml]: @ermish done

File: package.json (hunk 2 above)
Comment (RageZBla) [reply package.json]: @RageZBla redid the commits to touch only the GH workflows

**Comment by RageZBla** [comment]
@ermish would you be kind enough to take one more look? :bow:

File: .github/workflows/end-to-end.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+name: Tests: Cypress e2e
```
Comment (ermish) [inline .github/workflows/end-to-end.yml:1]: Sorry, can you also update the file name as well?

File: .github/workflows/unit-test.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Jest is JavaScript Testing Framework
```
Comment (ermish) [inline .github/workflows/unit-test.yml:1]: Here too

**Review by ermish — CHANGES_REQUESTED** [review]

**Comment by ermish** [comment]
> @ermish would you be kind enough to take one more look? :bow:

Looks good! Just need to update the filenes and we're good to go 🍞

File: .github/workflows/end-to-end.yml (hunk 3 above)
Comment (RageZBla) [reply .github/workflows/end-to-end.yml:1]: @ermish just making sure you want to use the toolname for the filename?

**Comment by ermish** [comment]
Sent a slack dm for this one

**Comment by theyokohamalife** [comment]
@RageZBla @ermish Hey! Not sure what's left to be done on this PR, but it's currently failing CI. Any update?

**Comment by RageZBla** [comment]
> @RageZBla @ermish Hey! Not sure what's left to be done on this PR, but it's currently failing CI. Any update?

I think the cypress failure is kind of expected, let me double check.

**Comment by RageZBla** [comment]
@theyokohamalife yes, this is due to the lack of tests. I can add a test if you prefer merging with the action green

**Comment by ermish** [comment]
Personally, I think we should add a dummy test, otherwise other contributors might think it's their code breaking.

**Comment by theyokohamalife** [comment]
> @theyokohamalife yes, this is due to the lack of tests. I can add a test if you prefer merging with the action green

@RageZBla That would be really helpful! 🙇🏾‍♀️

File: package.json [hunk 5]
```diff
@@ -10,7 +10,8 @@
         "lint:ci": "eslint . --config .eslintrc.js --ext .js,.jsx,.ts,.tsx,.vue --format @microsoft/eslint-formatter-sarif --output-file eslint-results.sarif",
         "prepare:ci": "nuxi prepare",
         "start": "nuxi start",
-        "test": "jest"
+        "test": "jest",
+        "e2e": "cypress open --e2e --browser chrome --port 9090"
```
Comment (ermish) [inline package.json]:
```suggestion
        "test:e2e": "cypress open --e2e --browser chrome --port 9090"
```

File: .github/workflows/cypress.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  push:
+    branches: [ "main", "ragezbla/20-github-actions" ] # TODO remove after checking it works
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '45 15 * * 4'
+
+jobs:
+  cypress-run:
```
[the remark this answers is not in the document]
Comment (ermish) [reply .github/workflows/cypress.yml:13]: @RageZBla  I think you missed this comment!

File: cypress.config.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { defineConfig } from 'cypress'
```
Comment (ermish) [inline cypress.config.ts:1]: I think these alerts will popup for every PR which is annoying. even if they're silenced for this one. 
In a separate PR, we can add `cypress.config.ts` to the ignore list if we can't get fix these

Comment (ermish) [reply cypress.config.ts:1]: updated

File: package.json (hunk 5 above)
Comment (ermish) [reply package.json]: updated

**Review by ermish — APPROVED** [review]
LGTM! 
@RageZBla thanks for all your help on this! Much appreciated 💯

File: .github/workflows/cypress.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (RageZBla) [reply .github/workflows/cypress.yml:13]: @ermish totally did 😬
